### PR TITLE
feat: support yarn berry

### LINKED ISSUE #24: Support yarn v2+
I noticed a cache issue with yarn v2+:

https://github.com/nrwl/ci/blob/38a699623acaa7c2887edc5f7739cbb0388bc038/.github/workflows/nx-cloud-agents.yml#L123-L126

 `yarn cache dir` doesn't exist in yarn 2+

This should be done instead: 

https://github.com/actions/cache/blob/main/examples.md#node---yarn-2

Edit: also `--immutable` should be used instead of `--frozen-lockfile` at the install step

### PR DESCRIPTION
**Pull request by Michsior14**
Super simple implementation for `yarn berry`. Feel free to take parts that you are happy with :)

Resolves #24.

### CODE REVIEW

File: .github/workflows/nx-cloud-agents.yml [hunk 1]
```diff
@@ -128,20 +129,34 @@ jobs:
...
+        if: steps.package_manager.outputs.name == 'yarn' && !startsWith(steps.versions.outputs.yarn_version, '1.')
+        id: yarn-berry-cache-dir-path
+        run: echo "dir=$(yarn config get cacheFolder)" >> $GITHUB_OUTPUT
+
+      - name: Use the node_modules cache if available [yarn berry]
+        if: steps.package_manager.outputs.name == 'yarn' && !startsWith(steps.versions.outputs.yarn_version, '1.')
+        uses: actions/cache@v3
+        with:
+          path: ${{ steps.yarn-berry-cache-dir-path.outputs.dir }}
+          key: ${{ runner.os }}-yarn-${{ hashFiles('**/yarn.lock') }}
```
Comment (JamesHenry) [inline .github/workflows/nx-cloud-agents.yml]: please include the node version in the cache key like we do in all other cases

**Review by JamesHenry — CHANGES_REQUESTED** [review]
Thanks a lot for this @Michsior14 - please also add an integration test workflow which leverages this. There is plenty of prior art for you to take inspiration from

File: .github/workflows/nx-cloud-agents.yml (hunk 1 above)
Comment (Michsior14) [reply .github/workflows/nx-cloud-agents.yml]: AFIK this is not needed for yarn berry: https://github.com/actions/setup-node/issues/325#issuecomment-916902678

Comment (JamesHenry) [reply .github/workflows/nx-cloud-agents.yml]: It's not _needed_ for any of them, but it is much better :)

Comment (JamesHenry) [reply .github/workflows/nx-cloud-agents.yml]: You don't change the node version of your project very often, but when you do it is most definitely worth busting caches over

**Comment by Michsior14** [comment]
> Thanks a lot for this @Michsior14 - please also add an integration test workflow which leverages this. There is plenty of prior art for you to take inspiration from

Will do it after work hours ;)

**Comment by Michsior14** [comment]
@JamesHenry Integration tests added.

**Comment by JamesHenry** [comment]
Looks like it's working well:

image

image

I ran it a second time to make sure there would be a cache hit and there was:

image

image

**Comment by JamesHenry** [comment]
@Michsior14 thanks a lot for this, please bump the minor version in the root `package.json` to `0.10.0` so that an appropriate release will be triggered post-merge

**Review by JamesHenry — APPROVED** [review]
LGTM 🎉 

I requested review from @meeroslav on this one as well as I still very jet lagged and I might have missed something 😄

**Comment by Michsior14** [comment]
Hey @meeroslav! Any update on the review? :)

File: .github/workflows/ci-integration-test-yarn-berry.yml [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+  cancel-in-progress: true
+
+jobs:
+  main:
+    name: Nx Cloud - Main Job
+    uses: ./.github/workflows/nx-cloud-main.yml
+    with:
+      working-directory: ./integration-tests/yarn-berry
+      parallel-commands: |
+        npx nx workspace-lint
```
Comment (meeroslav) [inline .github/workflows/ci-integration-test-yarn-berry.yml]: I would prefer having here `yarn nx...` over `npx nx...` but it's just nit-picking.

**Review by meeroslav — APPROVED** [review]
Looks good to me! Thank you

I left one minor comment, but it's not a blocker

**Pull request merged**